src/libraries/System.Diagnostics.Process/tests/ProcessWaitingTests.cs

```diff
@@ -82,6 +154,45 @@ public void SingleProcess_TryWaitMultipleTimesBeforeCompleting()
             Assert.True(p.WaitForExit(0));
         }
 
+        [Fact]
+        public async Task SingleProcess_TryWaitAsyncMultipleTimesBeforeCompleting()
+        {
+            Process p = CreateProcessLong();
+            p.Start();
+
+            // Verify we can try to wait for the process to exit multiple times
+            using (var cts = new CancellationTokenSource(0))
+            {
+                var token = cts.Token;
+                for (var i = 0; i < 2; i++)
+                {
+                    var ex = await Assert.ThrowsAnyAsync<OperationCanceledException>(() => p.WaitForExitAsync(token));
+                    Assert.Equal(token, ex.CancellationToken);
+                    Assert.False(p.HasExited);
+                }
+            }
+
+            // Then wait until it exits and concurrently kill it.
+            // There's a race condition here, in that we really want to test
+            // killing it while we're waiting, but we could end up killing it
+            // before hand, in which case we're simply not testing exactly
+            // what we wanted to test, but everything should still work.
+#pragma warning disable CS4014 // Because this call is not awaited, execution of the current method continues before the call is completed
+            Task.Delay(10).ContinueWith(_ => p.Kill());
```

Comment:
```suggestion
            _ = Task.Delay(10).ContinueWith(_ => p.Kill());
```

you can avoid the warnings using discard.

Reply:
Good to know! Fixed!